From fb1f95bd296f6185b894cbe6838e73cf310aefb7 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 5 Sep 2018 09:36:15 +0900
Subject: [PATCH 357/909] media: rcar-vin: Add FIS bit check before reading
 VnMS register

commit 3fe6064d48b47e28d786f89698f67ee8ac608f10 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

According to the latest H/W manual, following description is added.
"When video capture is operating, this bit should be read after
the FIS bit in VnINTS is set to 1." This patch adds its procedure.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/media/platform/rcar-vin/rcar-dma.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/rcar-vin/rcar-dma.c b/drivers/media/platform/rcar-vin/rcar-dma.c
index 3e6ddba52a63..0e50dc21f4ff 100644
--- a/drivers/media/platform/rcar-vin/rcar-dma.c
+++ b/drivers/media/platform/rcar-vin/rcar-dma.c
@@ -1180,6 +1180,10 @@ static irqreturn_t rvin_irq(int irq, void *data)
 	if (!rvin_seq_field_done(vin))
 		goto done;
 
+	/* Check FIS bit before reading VnMS register */
+	if (!(int_status & VNINTS_FIS))
+		goto done;
+
 	/* Prepare for capture and update state */
 	vnms = rvin_read(vin, VNMS_REG);
 	slot = (vnms & VNMS_FBS_MASK) >> VNMS_FBS_SHIFT;
-- 
2.17.1

